data/data/agent/files/usr/local/bin/load-config-iso.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+set -e
+
+AGENT_CONFIG_IMAGE_LABEL="agent_configimage"
+AGENT_CONFIG_ARCHIVE_FILE="config.gz"
+AGENT_CONFIG_MOUNT="/mnt/config_image"
+RENDEZVOUS_HOST_ENV="/etc/assisted/rendezvous-host.env"
+CLUSTER_IMAGE_SET="/etc/assisted/manifests/cluster-image-set.yaml"
+
+copy_archive_contents() {
+ 
+    # Mount device
+    mkdir -p ${AGENT_CONFIG_MOUNT}
+    mount "${device}" ${AGENT_CONFIG_MOUNT}
+
+    if [[ ! -f ${AGENT_CONFIG_MOUNT}/${AGENT_CONFIG_ARCHIVE_FILE} ]]; then
+       echo "Could not find file ${AGENT_CONFIG_ARCHIVE_FILE} on ${device}"
+       cleanup_files
+       return 1
+    fi
+
+    cp ${AGENT_CONFIG_MOUNT}/${AGENT_CONFIG_ARCHIVE_FILE} /tmp
+    gunzip -f /tmp/${AGENT_CONFIG_ARCHIVE_FILE}
+    unzipped_file=$(echo /tmp/${AGENT_CONFIG_ARCHIVE_FILE} | cut -d'.' -f1)
+
+    # Get the releaseImage in the archive and verify it matches the current cluster-image-set
+    release_image=$(grep releaseImage ${CLUSTER_IMAGE_SET} | sed -n -e 's/^.*releaseImage: //p')
+    arch_release_image=$(cpio -icv --to-stdout ${CLUSTER_IMAGE_SET} < "${unzipped_file}" | grep releaseImage | sed -n -e 's/^.*releaseImage: //p')
+    if [[ "${release_image}" != "${arch_release_image}" ]]; then
```

Comment:
Off the top of my head, I can't think of a reason not to compare the full contents of the files, since they're both supposed to be generated by the same version of the installer.

```
if diff "${CLUSTER_IMAGE_SET}" <(cpio -icv --to-stdout "${CLUSTER_IMAGE_SET}" <"${unzipped_file}"); then
```

Reply:
Changed to compare file.